src/parser/DASHTree.cpp

```diff
@@ -599,6 +599,25 @@ static void XMLCALL start(void* data, const char* el, const char** attr)
             attr += 2;
           }
         }
+        else if (strcmp(el, "SupplementalProperty") == 0)
+        {
+          const char *schemeIdUri(0), *value(0);
+
+          for (; *attr;)
+          {
+            if (strcmp((const char*)*attr, "schemeIdUri") == 0)
+              schemeIdUri = (const char*)*(attr + 1);
+            else if (strcmp((const char*)*attr, "value") == 0)
+              value = (const char*)*(attr + 1);
+            attr += 2;
+          }
+
+          if (schemeIdUri && value)
+          {
+            if (strcmp(schemeIdUri, "urn:mpeg:dash:adaptation-set-switching:2016") == 0)
+              dash->current_adaptationset_->switching_ids_ = split(value, ',');
+          }
+        }
```

Comment:
Fully qualified calls to functions please, e.g. `std::strcmp`.

Also, C++ casts only.

Reply:
i was just following the same code used above and below this
Can you please confirm you still want it changed?
Keep the same and then do a big refactor on all later?
Or fix as we go?

Reply:
i was just following the same code used above and below this
Can you please confirm you still want it changed?
Keep the same and then do a big refactor on all later?
Or fix as we go?

Reply:
If it’s fixed as we go it’s less work for the big refactor. But @glennguy’s call as it will fall to him.

Reply:
yeh, i'm just worried if he does a search and replace, having 1x different with a bunch of others may lead to a bad "replace"

Reply:
I'll sort these as part of the big refactor, for now lets get it in :)